rllib/evaluation/collectors/agent_collector.py

```diff
@@ -120,6 +125,32 @@ def is_empty(self) -> bool:
         """Returns True if this collector has no data."""
         return not self.buffers or all(len(item) == 0 for item in self.buffers.values())
 
+    def _check_view_requirement(self, vr_name: str, data: TensorType):
+        """Raises an AssertionError if data does not fit all view requirements that
+        have a view on data_col. Excludes ENV_ID that don't have a ViewRequirements."""
+
+        if (
+            log_once(f"view_requirement_{vr_name}_checked_in_agent_collector")
+            and vr_name in self.view_requirements
+        ):
+            vr = self.view_requirements[vr_name]
+            # We only check for the shape here, because conflicting dtypes are often
+            # because of float conversion
+            # TODO (Artur): Revisit rock_paper_scissors_multiagent and
+            #  test_multi_agent_env for cases where we accept a space that is not a
+            #  gym.Space
+            if hasattr(vr.space, "shape") and not vr.space.shape == np.shape(data):
```

Comment:
rock_paper_scissors_multiagent and test_multi_agent_env both produce spaces that are not gym Spaces but simply the state. We should revisit these because we currently explicity allow this (edge case?) in our init view requirements function.

Reply:
just curious, what is it, if it's not a gym.Space?